docs/content/en/docs/crd-ref/KeptnTaskDefinition/index.md

```diff
@@ -0,0 +1,127 @@
+---
+title: KeptnTaskDefinition
+description: Define tasks that can be run pre- or post-deployment
+weight: 89
+---
+
+A `KeptnTaskDefinition` is a CRD used to define tasks
+that can be run by the Keptn Lifecycle Toolkit
+as part of pre- and post-deployment phases of a deployment.
+The task definition is a [Deno](https://deno.land/) script.
+Please, refer to the [function runtime](https://github.com/keptn/lifecycle-toolkit/tree/main/functions-runtime)
+for more information about the runtime.
+In the future, we also intend to support other runtimes,
+especially running a container image directly.
+
+## Synopsis
+
+```yaml
```

Comment:
we have code examples of our CRDs already prepared (mostly). I would use a link here instead of actual code.
You can find the examples here: https://github.com/keptn/lifecycle-toolkit/issues/888

Reply:
I would love to embed this content in the source and have it autogenerated!  But I figured I needed to hand-write a few pages to work things out and then we know what we want the tools to do.

Reply:
I would very much like the "Synopsis" to show on the reference page that is displayed in the documentation, with a format similar to what I have here.  But it would be ideal if that were pulled from the software source so it does not need to be maintained manually!  I very much like the idea of just pointing to existing examples -- that allows people to see how this CRD fits in the "big picture" as it were.